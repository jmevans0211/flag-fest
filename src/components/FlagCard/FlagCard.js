import React from 'react';
import { connect } from 'react-redux';
import './FlagCard.scss'


export const FlagCard = ({ countries, handleGuess, flagsGuessed }) => {
  let name;
  let flag;
  let wrongAnswer;

  
  if (countries[0]) {
    name = countries[0].name
    flag = countries[0].flag
  }

  const wrongAnswers = countries.map(country => {
    return country.name
  });

  if (countries.length > 1) {
    do {
      wrongAnswer = wrongAnswers[Math.floor(Math.random() * wrongAnswers.length)];
    } while (wrongAnswer === name)
  } else {
    wrongAnswer = flagsGuessed[Math.floor(Math.random() * flagsGuessed.length)].name;
  }

  return (
    <section className="flag-card-container">
      <img src={flag} />
      <button onClick={() => handleGuess('correct')}>{name}</button>
      <button onClick={() => handleGuess('incorrect')}>{wrongAnswer}</button>
    </section>
  )
}

export const mapStateToProps = state => ({
  countries: state.countries,
});

export default connect(mapStateToProps, null)(FlagCard);
