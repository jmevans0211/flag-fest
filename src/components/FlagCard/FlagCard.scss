.flag-card-container {
  border: 1px solid purple;
}