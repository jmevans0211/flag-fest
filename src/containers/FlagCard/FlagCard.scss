$button-color-next-page: #00E1FF;

// .flag-card-container {
//   margin-top: 30px;
//   display: grid;
//   grid-template-columns: 800px 800px;
//   grid-template-rows: auto auto;
//   grid-template-areas: "card-flag card-flag"
//                        "positionA positionB"                    
// }

.answer-container {
  margin-top: 30px;
  display: grid;
  grid-template-columns: 800px 800px;
  grid-template-rows: auto;
  grid-template-areas: "positionA positionB"   
}

.positionA {
  grid-area: positionA;
}

.positionB {
  grid-area: positionB;
}

.card-flag {
  // grid-area: card-flag;
  width: 700px;
}

h6 {
  background-color: $button-color-next-page;
  border-radius: 10px;
  color: white;
  cursor: pointer;
  font-size: 34px;
  margin: 40px auto;
  padding: 25px 0;
  text-align: center;
  width: 70%;
}