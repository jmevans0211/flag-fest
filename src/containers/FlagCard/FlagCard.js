import React from 'react';
import { connect } from 'react-redux';
import './FlagCard.scss'


export const FlagCard = ({ countries, handleGuess, flagsGuessed, correctClass, wrongClass }) => {
  let name;
  let flag;
  let wrongAnswer;

  
  if (countries[0]) {
    name = countries[0].name
    flag = countries[0].flag
  }

  const wrongAnswers = countries.map(country => {
    return country.name
  });

  if (countries.length > 1) {
    do {
      wrongAnswer = wrongAnswers[Math.floor(Math.random() * wrongAnswers.length)];
    } while (wrongAnswer === name)
  } else {
    wrongAnswer = flagsGuessed[Math.floor(Math.random() * flagsGuessed.length)].name;
  }

  return (
    <section className="flag-card-container">
      <img className='card-flag' src={flag} />
        <h6 className={correctClass} onClick={() => handleGuess('correct')} role="button">{name}</h6>
        <h6 className={wrongClass} onClick={() => handleGuess('incorrect')} role="button">{wrongAnswer}</h6>
    </section>
  )
}

export const mapStateToProps = state => ({
  countries: state.countries,
});

export default connect(mapStateToProps, null)(FlagCard);
