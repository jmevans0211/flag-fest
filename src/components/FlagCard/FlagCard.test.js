import React from 'react';
import { shallow } from 'enzyme';
import { FlagCard } from './FlagCard';

describe ('FlagCard', () => {
  let wrapper;
  beforeEach(() => {
    let mockHandleGuess = jest.fn()
    wrapper = shallow(<FlagCard 
      countries={
        [
          {
            flag: "https://restcountries.eu/data/srb.svg",
            name: "Serbia",
            numericCode: 688,
            region: "Europe"
          },
          {
            flag: "https://restcountries.eu/data/lux.svg",
            name: "Luxembourg",
            numericCode: 442,
            region: "Europe"
          }
        ]
      }
      handleGuess={mockHandleGuess}
    />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });



});