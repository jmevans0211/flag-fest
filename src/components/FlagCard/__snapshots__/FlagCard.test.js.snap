// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagCard should match snapshot 1`] = `
<section
  className="flag-card-container"
>
  <img
    src="https://restcountries.eu/data/srb.svg"
  />
  <button
    onClick={[Function]}
  >
    Serbia
  </button>
  <button
    onClick={[Function]}
  >
    Genovia
  </button>
</section>
`;
